rfcs/text/0032-structured-benchmark-programs.md

```diff
@@ -0,0 +1,224 @@
+| X-Ray Absorption Spectroscopy                 | ✔️ | ❌ | ✔️ | ❓ | ❓ | ❌ | ❌ | ❌ | [Paper](https://arxiv.org/abs/2405.11015), [Tutorial](https://pennylane.ai/qml/demos/tutorial_xas) | ❌ | ✔️ |
+| Repeat-Until-Success                          | ❌ | ✔️ | ❌ | ❌ | ❌ | ✔️ | ❌ | ❌ | [Paper](https://arxiv.org/abs/1311.1074) | ❌ | ❌ |
+| Quantum Metropolis Sampling                   | ❌ | ✔️ | ✔️ | ❓ | ❓ | ✔️ | ❌ | ❌ | [Paper](https://arxiv.org/abs/0911.3635) | ❌ | ✔️ |
+| ML-QAE                                        | ❌ | ✔️ | ✔️ | ❓ | ❓ | ✔️ | ❌ | ❌ | [Paper](https://arxiv.org/abs/1904.10246) | ❌ | ✔️ |
+| Quantum Multiplexers                          | ✔️ | ❌ | ✔️ | ✔️ | ✔️ | ❌ | ❌ | ❌ | [Paper](https://arxiv.org/abs/quant-ph/0410066) | ✔️ | ✔️ |
+| Toffoli-heavy Circuits                        | ✔️ | ❌ | ✔️ | ❌ | ❌ | ❌ | ✔️ | ✔️ | [Paper](https://arxiv.org/abs/1904.01671) | ✔️ | ❌ |
+| Magic State Distillation                      | ✔️ | ✔️ | ✔️ | ❌ | ❌ | ✔️ | ❌ | ❌ | [Paper](https://arxiv.org/abs/quant-ph/0403025) | ✔️ | ✔️ |
+| Logical State Preparation                     | ✔️ | ✔️ | ✔️ | ❌ | ❌ | ✔️ | ❌ | ❌ | [Nielsen and Chuang](https://doi.org/10.1017/CBO9780511976667) | ✔️ | ✔️ |
+| Syndrome Measurement and Correction           | ✔️ | ✔️ | ✔️ | ❌ | ❌ | ✔️ | ❌ | ❌ | [Nielsen and Chuang](https://doi.org/10.1017/CBO9780511976667) | ✔️ | ✔️ |
+| QAOA with Fixed Repetitions                   | ✔️ | ❌ | ❌ | ✔️ | ❌ | ❌ | ❌ | ❌ | [Paper](https://arxiv.org/abs/1411.4028) | ❌ | ❌ |
+| VQE Ansatz with Fixed Repetitions             | ✔️ | ❌ | ❌ | ✔️ | ❌ | ❌ | ❌ | ❌ | [Paper](https://arxiv.org/abs/1304.3061) | ✔️ | ❌ |
+| VQE                                           | ✔️ | ❌ | ❌ | ✔️ | ❌ | ✔️ | ❌ | ❌ | [Paper](https://arxiv.org/abs/1304.3061) | ✔️ | ✔️ |
+
+### Symbol Legend
+
+| Symbol | Description |
+|--------|-------------|
+| ✔️ | Feature is required |
+| ❌ | Feature is not used |
+| 🟦 | Feature may be used depending on specific implementation |
+| ❓ | Requires some further research |
+
+### Category Details
+
+| Category | Description |
+|----------|-------------|
+| statically-bounded loops                      | Program uses loops with a constant number of repetitions. |
+| dynamically-bounded loops                     | Loop bounds depend on value calculated at runtime. |
+| dynamic qubit indexing                        | Gates are applied to qubits with non-constant indices (e.g. loop variable). |
+| dynamic classical values                      | Gates use other classical values computed at runtime (e.g. rotation angles taken from arrays). |
+| conditionals on originally classical values   | Conditional blocks are used where the condition depends on values that were *not* measurement results. |
+| conditionals on measurement results           | Conditional blocks are used where the condition depends on values that depend on measurement results. |
```

Comment:
I'm curious for the conditional we separate out classical dynamic values from "quantum dynamic values" (i.e. measurement results), but we didn't do this for the loops (I think both are contained in the dynamically-bounded loops).